### PR TITLE
update app_build.sh to automatically detect some HPC platforms

### PR DESCRIPTION
**Pull request by guoqing-noaa**
## DESCRIPTION OF CHANGES: 

- update app_build.sh to automatically detect some HPC platforms

## TESTS CONDUCTED: 

- On machines/platforms:

- [ ] WCOSS2
- [ ] Hera
- [ ] Orion
- [x] Jet

- Test cases: 

- [x] Non-DA engineering test
- [ ] DA engineering test
- [ ] Other sample scripts:

## ISSUE: 

- Fixes the issue(s) mentioned in #81

### CODE REVIEW

**Comment by MatthewPyle-NOAA** [comment]
@chan-hoo are you good with it now?  Looks safe to me.

File: sorc/app_build.sh [hunk 1]
```diff
@@ -202,9 +202,13 @@ EXTERNALS="${EXTERNALS^^}"
 
 # check if PLATFORM is set
 if [ -z $PLATFORM ] ; then
-  printf "\nERROR: Please set PLATFORM.\n\n"
-  usage
-  exit 0
+  # Automatically detect HPC platforms for wcoss2, hera, jet, orion, etc
+  source ../ush/fix_rrfs_locations.sh
```
Comment (chan-hoo) [inline sorc/app_build.sh:206]: @guoqing-noaa, can you change this relative path to the absolute path like: `source ${HOME_DIR}/ush/fix_rrfs_locations.sh`?

Comment (guoqing-noaa) [reply sorc/app_build.sh]: Done. Thanks!

**Comment by chan-hoo** [comment]
@guoqing-noaa, wow! so fast! I am testing this PR now.

**Comment by chan-hoo** [comment]
I confirm this works correctly with and without the -p argument. Thanks for this change, @guoqing-noaa! Approving.

**Comment by guoqing-noaa** [comment]
> I confirm this works correctly with and without the -p argument. Thanks for this change, @guoqing-noaa! Approving.

Thanks for the testing! @chan-hoo

**Review by MatthewPyle-NOAA — APPROVED** [review]
Looks good to me, and thanks for cleaning up the typos.

**Pull request merged**